lib/spack/docs/known_issues.rst

```diff
@@ -0,0 +1,38 @@
+.. Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+============
+Known Issues
+============
+
+This is a list of known issues in Spack. It provides ways of getting around these
+problems if you encounter them.
+
+------------------------------------------------
+Spack does not seem to respect ``packages.yaml``
+------------------------------------------------
+
+A common problem in Spack v0.18 and above is that package, compiler and target
+preferences specified in ``packages.yaml`` do not seem to be respected. Spack picks the
+"wrong" compilers and their versions, package versions and variants, and
+micro-architectures.
+
+This is however not a bug. In order to reduce the number of builds of the same
+packages, the concretizer values reuse of installed packages higher than preferences
+set in ``packages.yaml``. Note that ``packages.yaml`` specifies only preferences, not
+hard constraints.
+
+To verify you are affected by this problem, you can use
+``spack spec --install-status <spec>``. This should indicate with a ``[+]`` that
+certain "problematic" specs were picked because they were already installed.
```

Comment:
This might also happen with buildcaches, in which case the user won't see a `[+]` :thinking:

Reply:
We should add (in another PR, of course) a new symbol, for instance `[B]`, to indicate that a spec has been selected from a buidcache.

Reply:
I'll just remove that line